File: src/NodeParser/NullLiteralNodeParser.ts
import ts from "typescript";
import type { Context } from "../NodeParser.js";
import type { SubNodeParser } from "../SubNodeParser.js";
import type { BaseType } from "../Type/BaseType.js";
import { NullType } from "../Type/NullType.js";

export class NullLiteralNodeParser implements SubNodeParser {
    public supportsNode(node: ts.NullLiteral): boolean {
        return node.kind === ts.SyntaxKind.NullKeyword;
    }
    public createType(node: ts.NullLiteral, context: Context): BaseType {
        return new NullType();
    }
}

File: internal/parser/null_literal_node_parser.go
package parser

import (
	"github.com/microsoft/typescript-go/shim/ast"

	"github.com/vega/ts-json-schema-generator/internal/types"
)

// NullLiteralNodeParser handles the `null` literal
// (src/NodeParser/NullLiteralNodeParser.ts).
type NullLiteralNodeParser struct{}

func NewNullLiteralNodeParser() *NullLiteralNodeParser {
	return &NullLiteralNodeParser{}
}

func (p *NullLiteralNodeParser) SupportsNode(node *ast.Node) bool {
	return node.Kind == ast.KindNullKeyword
}

func (p *NullLiteralNodeParser) CreateType(_ *ast.Node, _ *Context, _ *types.ReferenceType) types.Type {
	return &types.NullType{}
}
